src/kernel/mbrowse.ml

```diff
@@ -221,6 +221,22 @@ let of_signature sg =
   in
   [env, Browse_raw.Signature sg]
 
+let rec of_structure_items lst =
+  match lst with
+  | [] -> []
+  | (env,head) :: tl -> 
+    Browse_raw.Structure_item (head,env) :: (of_structure_items tl)
```

Comment:
I think that the function could be tail-recursive?  (but in fact it seems that it is `map` and it will probably do not require a `List.rev` on the previous file?): 

```ocaml
let of_structure_items = 
  List.map (fun (head, env) -> 
       Browse_raw.structure_item (head, env)) 
```